jackson/src/main/scala/com/lightbend/lagom/internal/jackson/JacksonJsonSerializer.scala

```diff
@@ -87,14 +86,16 @@ private[lagom] class JacksonJsonSerializer(val system: ExtendedActorSystem)
     val startTime = if (isDebugEnabled) System.nanoTime else 0L
     val compressed = isGZipped(bytes)
 
-    val migration = migrations.get(manifest)
+    val (fromVersion, manifestClassName) = parseManifest(manifest)
+
+    val migration = migrations.get(manifestClassName)
```

Comment:
@rcavalcanti I think setting up migrations for specific version of the same class was a side effect of the bug this PR is trying to fix.

_But_, if you have that case, you may add `if(versionFrom== 3) new TestEventMigration2()...`inside `TestEventMigration`.